core/lib/object_store/src/gcs.rs

```diff
@@ -123,6 +138,17 @@ fn has_transient_io_source(err: &(dyn StdError + 'static)) -> bool {
     get_source::<io::Error>(err).is_some()
 }
 
+/// This is necessary due to the fact that Acquiring a semaphore can fail if the semaphore is closed.
+/// Whilst current code does not expect such scenario, nor "should it be possible", it is better to have a graceful catch case instead of unwrapping.
+impl From<AcquireError> for ObjectStoreError {
+    fn from(err: AcquireError) -> Self {
+        ObjectStoreError::Other {
+            source: err.into(),
+            is_retriable: true,
```

Comment:
I understand the reasoning behind not unwrapping, but why is this error retriable? If the semaphore is closed, it'll remain closed, so all subsequent requests to the store will fail.